File: Data Structures & Algorithms/concatenation-of-array/submission-3.java
class Solution {
    public int[] getConcatenation(int[] nums) {
        int[] concat_array = new int[nums.length * 2];

        for (int i = 0; i < nums.length; i++) {
            concat_array[i] =  nums[i];
            concat_array[i + nums.length] = nums[i];
        }

        return concat_array;
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-3.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_set<int> unique_nums;

        for (int num : nums) {
            if (unique_nums.find(num) != unique_nums.end()){
                return true;
            } else {
                unique_nums.insert(num);
            }
        }
        
        return false;
    }
};